OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs

```diff
@@ -0,0 +1,334 @@
+}";
+
+            var requestUri = string.Format("{0}/odata/Customers(SSN='special-SSN')", this.BaseAddress);
+            HttpResponseMessage response = await Client.GetAsync(requestUri);
+
+            response.EnsureSuccessStatusCode();
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(expect, responseContent);
+        }
+
+        public static TheoryDataSet<string, string> SingleAlternateKeysCases
+        {
+            get
+            {
+                var data = new TheoryDataSet<string, string>();
+                for (int i = 1; i <= 5; i++)
+                {
+                    data.Add("Customers(" + i + ")", "Customers(SSN='SSN-" + i + "-" + (100 + i) + "')");
+                }
+
+                return data;
+            }
+        }
+
+        [Theory]
+        [PropertyData("SingleAlternateKeysCases")]
+        public async Task EntityWithSingleAlternateKeys_ReturnsSame_WithPrimitiveKey(string declaredKeys, string alternatekeys)
+        {
+            // query with declared key
+            var requestUri = string.Format("{0}/odata/{1}", this.BaseAddress, declaredKeys);
+            HttpResponseMessage response = await Client.GetAsync(requestUri);
+
+            response.EnsureSuccessStatusCode();
+            string primitiveResponse = await response.Content.ReadAsStringAsync();
+
+            // query with alternate key
+            requestUri = string.Format("{0}/odata/{1}", this.BaseAddress, alternatekeys);
+            response = await Client.GetAsync(requestUri);
+
+            response.EnsureSuccessStatusCode();
+            string alternatekeyResponse = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(primitiveResponse, alternatekeyResponse);
+        }
+
+        [Fact]
+        public async Task QueryEntityWithMultipleAlternateKeys_Returns_SameEntityWithPrimitiveKey()
+        {
+            // query with declared key
+            var requestUri = string.Format("{0}/odata/Orders(2)", this.BaseAddress);
+            HttpResponseMessage response = await Client.GetAsync(requestUri);
+
+            response.EnsureSuccessStatusCode();
+            string primitiveResponse = await response.Content.ReadAsStringAsync();
+
+            // query with one alternate key
+            requestUri = string.Format("{0}/odata/Orders(Name='Order-2')", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string nameResponse = await response.Content.ReadAsStringAsync();
+
+            // query with another alternate key
+            requestUri = string.Format("{0}/odata/Orders(Token=75036B94-C836-4946-8CC8-054CF54060EC)", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string guidRespoonse = await response.Content.ReadAsStringAsync();
```

Comment:
Convert it to tokenResponse to have same naming convention as nameResponse. Or change both to tokenGuidResponse, nameStringResponse.